Task2.3/Program.cs

```diff
@@ -0,0 +1,104 @@
+﻿class Entity
+{
+    protected string _textDescription;
+
+    public Entity(string textDescription)
+    {
+        _textDescription = textDescription;
+    }
+}
+
+abstract class Lesson : Entity
+{
+    public Lesson(string textDescription) : base(textDescription) { }
+
+    public abstract Lesson Clone();
+}
+
+class Lecture : Lesson
+{
+    private string _topic;
+
+    public Lecture(string textDescription, string topic) : base(textDescription)
+    {
+        _topic = topic;
+    }
+
+    public override Lesson Clone()
+    {
+        return new Lecture(_textDescription, _topic);
+    }
+}
+
+class PracticalLesson : Lesson
+{
+    private string _taskCondition;
+    private string _solution;
+
+    public PracticalLesson(string textDescription, string taskCondition, string solution) : base(textDescription)
+    {
+        _taskCondition = taskCondition;
+        _solution = solution;
+    }
+
+    public override Lesson Clone()
+    {
+        return new PracticalLesson(_textDescription, _taskCondition, _solution);
+    }
+}
+
+class Training
+{

```

Comment:
It assumes that now you do not now about List, and has to use simple array ))). With resize after Add, but OK )